libraries/libcore/inc/log.h

```diff
@@ -32,20 +32,18 @@ typedef enum {
   } while (0)
 #else
 #include <stdbool.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include "store.h"
 
-#define MAX_LOG_LEN 256
-static char s_log_buf[MAX_LOG_LEN];
-
-#define LOG(level, fmt, ...)                                                                  \
-  do {                                                                                        \
-    if ((level) >= LOG_LEVEL_VERBOSITY) {                                                     \
-      memset(s_log_buf, 0, sizeof(s_log_buf));                                                \
-      int len = snprintf(s_log_buf, sizeof(s_log_buf), "[%u] %s:%u: " fmt, (level), __FILE__, \
-                         __LINE__, ##__VA_ARGS__);                                            \
-      log_export(s_log_buf, len);                                                             \
-    }                                                                                         \
+#define LOG(level, fmt, ...)                                                                    \
+  do {                                                                                          \
+    if ((level) >= LOG_LEVEL_VERBOSITY) {                                                       \
+      char *log;                                                                                \
+      int len = asprintf(&log, "[%u] %s:%u: " fmt, (level), __FILE__, __LINE__, ##__VA_ARGS__); \
+      log_export(log, len);                                                                     \
```

Comment:
Does this account for when len==-1 (ie if memory allocation fails)?

Reply:
It doesn't, but also memory allocation should basically never fail on x86---I'll add a check regardless just in case.

Reply:
Alright so adding the check makes the `mu_init_conds` test deadlock for unknown reasons I don't want to look into, and if memory allocation fails on x86 we have bigger problems, so I think it's fine to merge without the check.